lib/services/autofs.pm

```diff
@@ -0,0 +1,100 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Package for autofs service tests
+#
+# Maintainer: Antonio Caristia <acaristia@suse.com>, Huajian Luo <hluo@suse.com>
+
+package services::autofs;
+use base "consoletest";
+use testapi;
+use strict;
+use warnings;
+use version_utils qw(is_sle is_jeos);
+use autofs_utils qw(setup_autofs_server check_autofs_service);
+use utils qw(systemctl zypper_call);
+
+sub install_service {
+    zypper_call("in autofs");
+}
+
+sub enable_service {
+    systemctl 'enable autofs';
+}
+
+sub start_service {
+    systemctl 'start autofs';
+}
+
+# check service is running and enabled
+sub check_service {
+    systemctl 'is-enabled autofs.service';
+    systemctl 'is-active autofs';
+}
+
+sub check_function {
+    select_console 'root-console';
+
+    # mkisofs is not distributed in JeOS based on sle12
+    my $mk_iso_tool = (is_jeos and is_sle('<15')) ? 'genisoimage' : 'mkisofs';
+    zypper_call("in autofs $mk_iso_tool") if (is_sle('15+') or is_jeos);
+
+    my $autofs_conf_file       = '/etc/auto.master';
+    my $autofs_map_file        = '/etc/auto.master.d/autofs_regression_test.autofs';
+    my $test_conf_file         = '/etc/auto.iso';
+    my $test_mount_dir         = '/mnt/test_autofs_local';
+    my $file_to_mount          = '/tmp/test-iso.iso';
+    my $test_conf_file_content = "echo  iso     -fstype=auto,ro         :$file_to_mount > $test_conf_file";
+
+    assert_script_run("mkdir -p ${test_mount_dir}");
+    assert_script_run("chmod 0777 ${test_mount_dir}");
+    assert_script_run("dd if=/dev/urandom of=$test_mount_dir/README bs=4024 count=1");
+    assert_script_run("$mk_iso_tool -o $file_to_mount $test_mount_dir");
+    assert_script_run("ls -lh $file_to_mount");
+    assert_script_run("test -f $autofs_conf_file");
+
+    setup_autofs_server(autofs_conf_file => $autofs_conf_file,
+        autofs_map_file        => $autofs_map_file,
+        test_conf_file         => $test_conf_file,
+        test_conf_file_content => $test_conf_file_content,
+        test_mount_dir         => $test_mount_dir);
+
+    systemctl 'restart autofs';
+
+    assert_script_run("ls $test_mount_dir/iso");
+    my $mount_output_triggered = script_output("mount | grep -e $file_to_mount");
+    record_soft_failure("Something went wrong, target is already mounted") unless $mount_output_triggered =~ /$file_to_mount/;
```

Comment:
soft failure must add issue number, do not change the original code unless you encounter block issue here.